utils/clflags.mli

```diff
@@ -235,11 +235,13 @@ val insn_sched : bool ref
 val insn_sched_default : bool
 
 module Compiler_pass : sig
-  type t = Parsing | Typing
+  type t = Parsing | Typing | Scheduling
   val of_string : string -> t option
   val to_string : t -> string
   val passes : t list
-  val pass_names : string list
+  val pass_names : (t -> bool) -> bool -> string list
```

Comment:
The name `pass_names` does not feel optimal since the pass are filtered. Maybe more `available_pass_names`?
Adding a label to the boolean argument would be nice.
I would rather introduce the predicate function once there is an actual use case.

Reply:
I removed the predicate function argument from the interface, but it's still lurking in the module, because use it in the two follow-up PRs.  I simplified the interface following your suggestions above. The uses of this interfaces look simpler too, I think.